carbon-design-system/carbon-for-ibm-dotcom
# fix(masthead): add missing divider line

### Related Ticket(s)

GH Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6402
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2169

### Description

- This PR moves the styling of the divider line from `dds-top-nav-name` and adds it to `dds-masthead-logo` so that it's visible on variant where there's no `dds-top-nav-name` being defined.

### Testing Instruction

- Visit the default `Masthead` and `Cloud Masthead` - `/?path=/story/components-masthead--default` and `/?path=/story/components-cloud-masthead--default`
- Ensure that the divider line is visible and aligns with the design specs.

### Expected Display

Default Masthead without the platform name
![Screen Shot 2021-09-09 at 3 00 50 PM](https://user-images.githubusercontent.com/1815714/132747196-b51e6e6b-da53-42a8-a59f-5ac9dedd9e3d.png)

Cloud Masthead with the platform name
![Screen Shot 2021-09-09 at 3 01 08 PM](https://user-images.githubusercontent.com/1815714/132747146-b321d5db-a62f-4fa6-8b60-a366ff521735.png)
